packages/config/src/getConfig.ts

```diff
@@ -0,0 +1,124 @@
+import JsonFile from '@expo/json-file';
+import { formatExecError } from 'jest-message-util';
+import path from 'path';
+
+import { ConfigContext, ExpoConfig } from './Config.types';
+import { ConfigError } from './Errors';
+import { fileExists } from './Modules';
+
+// support all common config types
+export const allowedConfigFileNames: string[] = (() => {
+  const prefix = 'app';
+  return [
+    // order is important
+    // TODO: Bacon: Slowly rollout support for other config languages: ts, yml, toml
+    `${prefix}.config.js`,
+    `${prefix}.config.json`,
+    `${prefix}.config.json5`,
```

Comment:
With .js support I don't think we should encourage JSON5 more. I also think the cost of supporting YAML and TOML isn't worth it. Supporting TS might be worthwhile since it feels more consistent with the rest of a project.